### PR TITLE
[Feature] add hr advisor email

### LINKED ISSUE #9232: HR advisor email - optional field on the submit page for talent requests
### User story
As a request responder, I want to be able to CC the talent seeker's (manager's) HR advisor so can support the talent seeker and process the hire more efficiently.

### Details
On the submit page for the "find talent" flow, recruitment team has requested an additional optional field for the HR advisor's email. This would allow them to loop in the HR advisor when they send talent back to the talent seeker (manager).

### Acceptance Criteria
- [ ] New "HR advisor email" input field on the Find Talent submit request page.
- [ ] New field is optional.
- [ ] New field shows up on the request information in Admin side.

### Design files

[Figma link](https://www.figma.com/file/dDo0sUzYFyplp7DvlLHKYJ/Admin---Requests-details?type=design&node-id=318%3A554&mode=design&t=MxXDhgXiISEJgFQR-1) by @Jerryescandon 

<details><summary>Curent design (Request form page) </summary>
<p>

image

</p>
</details> 

<details><summary>New proposed design (Request form page) </summary>
<p>

image

</p>
</details> 

<details><summary>Current design (Admin request details page) </summary>
<p>

image

</p>
</details> 

<details><summary>New design (Admin request details page) </summary>
<p>

image

</p>
</details> 

## Acceptance Criteria
- [ ] hr advisor email added to model for talent requests (database, model, seeder, factory, schema)
- [ ] Can be set on form (optional)
- [ ] Can be seen on the admin side

### PR DESCRIPTION
**Pull request by brindasasi**
🤖 Resolves #9232.

## 👋 Introduction

Describe what this PR is solving.

## 🕵️ Details

Add any additional details that could assist with reviewing or testing this PR.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Go  to Find Talent , request candidates from a pool and land on the Search Request Summary Page
2. Observe the new field "HR advisor email" and verify you can successfully submit request. 
3. Confirm it is optional and takes email format 
4. Go to Admin page and open up the request you just submitted and confirm the field is still there. 
5. Verify the new field takes empty value as well

## 📸 Screenshot

Screenshot 2024-02-23 at 5 13 14 PM

Screenshot 2024-02-23 at 5 15 22 PM

### CODE REVIEW

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/ViewSearchRequest.tsx [hunk 1]
```diff
@@ -49,6 +49,7 @@ const ManagerInfo = ({
     department,
     email,
     managerJobTitle,
+    hrAdvisorEmail,
```
[the remark this answers is not in the document]
Comment (brindasasi) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/ViewSearchRequest.tsx:52]: fixed in [93becfa](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9495/commits/93becfa4bef042191c6ca35d949f107125c66a58)

File: api/database/migrations/2024_02_23_145241_add_hr_advisor_email_column_pool_candidate_search_requests.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('pool_candidate_search_requests', function (Blueprint $table) {
+            $table->text('hr_advisor_email')->nullable()->default(null);
```
[the remark this answers is not in the document]
Comment (brindasasi) [reply api/database/migrations/2024_02_23_145241_add_hr_advisor_email_column_pool_candidate_search_requests.php]: fixed in [93becfa](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9495/commits/93becfa4bef042191c6ca35d949f107125c66a58)

**Comment by brindasasi** [comment]
> Just a few suggestions but looking good so far! In addition to the code comments, we should probably add some fake data to the factory as well so some are seeded with the new field.

Thanks! I think there is only one as general search request for fake data.  .. let me know if I missed anything . fixed in [93becfa](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9495/commits/93becfa4bef042191c6ca35d949f107125c66a58)

**Comment by esizer** [comment]
> Thanks! I think there is only one as general search request for fake data. .. let me know if I missed anything . fixed in [93becfa](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9495/commits/93becfa4bef042191c6ca35d949f107125c66a58)

Nice! Just one more spot. We will need to add it to the factory as well. Remembering that we should also null it in the `withOldRequests` state method.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/fea94bbf85b882d2508d1161642dc4e5516ea4bd/api/database/factories/PoolCandidateSearchRequestFactory.php#L30-L45

**Pull request merged**